tests/system_tests/test_take_data.py

```diff
@@ -0,0 +1,47 @@
+from concurrent.futures import Future
+from pathlib import Path
+from typing import Any
+
+from blueapi.client.client import BlueapiClient
+from blueapi.service.model import TaskRequest
+from bluesky_stomp.messaging import MessageContext, StompClient
+from bluesky_stomp.models import MessageTopic
+
+
+def test_collect_data(
+    client: BlueapiClient,
+    stomp_client: StompClient,
+    latest_comissioning_instrument_session: str,
+    data_directory: Path,
+) -> None:
+    # Listen for NeXus file events, see below
+    nexus_finished_message = Future()
+
+    def on_nexus_message(message: dict[str, Any], _: MessageContext) -> None:
+        if message["status"] == "FINISHED":
+            nexus_finished_message.set_result(message)
+
+    stomp_client.subscribe(MessageTopic(name="gda.messages.scan"), on_nexus_message)
+
+    # Run plan
+    end_event = client.run_task(
+        TaskRequest(
+            name="count",
+            params={"detectors": ["sample_det"], "num": 5},
+            instrument_session=latest_comissioning_instrument_session,
+        ),
+        timeout=10.0,
+    )
```

Comment:
And then this is run by the live blueapi application?

Reply:
Yes, which is why I've reconfigured `pyproject.toml` so it doesn't run this when we run unit tests, so they can still work in CI. This is basically a test script you can run after deploying.